### PR TITLE
Add new always() function

### PR DESCRIPTION
**Pull request by seanpc9**
Add a function that allows a promise to be run after the completion of an initial promise regardless of success or failure of the initial promise.

### CODE REVIEW

**Comment by cschandler** [comment]
@seanpc9 And I had a discussion about the name. I'm not a huge fan of `always`, and would prefer the first promise to be the one siloed off, as it's result is the one that we don't care about. Something along the lines of `first`.

**Comment by mikerunnals** [comment]
Here, I think `promise.always` is similar to `promise.success` and `promise.recover`.

This `always` is reminiscent of `finally` in try/catch/finally (Java, C#, etc). 
Swift seems to have a `defer` for this ([see error handling docs](https://developer.apple.com/library/content/documentation/Swift/Conceptual/Swift_Programming_Language/ErrorHandling.html#//apple_ref/doc/uid/TP40014097-CH42-ID508)).

Perhaps `defer` or `finally` would be better?

File: Sources/Promise.swift [hunk 1]
```diff
@@ -294,6 +294,20 @@ public struct Promise<Value> {
...
+     - parameter resultTask: A function to be called that returns a promise.
+     - returns: A promise whose task runs this promise, and then calls the result task after it succeeds or fails.
+     */
+    public func always<U>(_ resultTask: @escaping () -> Promise<U>) -> Promise<U> {
+        return map { _ in () }.recover { _ in
+            Promise<()>(value: ())
+        }.flatMap { _ in
+            return resultTask()
+        }
+    }
```
Comment (kconner) [inline Sources/Promise.swift]: This transformation does one more thing that your documentation doesn't mention, which is that it discards the result of the first task. And as a result, it means step two's input has to be void. That makes me think this is too specific, and we want more elementary and generic operations, like:

1. Map to (), which I don't think belongs in the library
2. Recover, but ignoring the error and assuming a default value (`recover(value:)`?), which in this case could be `()`
3. Do the second task (this is already just `flatMap`)

In other words I think you could write:

```
public func recover(value: Value) -> Promise<Value> {
    return recover { _ in
        Promise(value: value)
    }
}
```

… and then in any application, `taskOne.recover(value: someDefaultValue).flatMap(taskTwo)`.

In your application that would mean `taskOne.map { _ in }.recover(value: ()).flatMap(taskTwo)`.

Comment (seanpc9) [reply Sources/Promise.swift]: @kconner this would solve my problem, but I don't think it makes enough of a difference that it is worth including. IMO I say we either update the documentation and potentially rename the `always` function or we just don't add it at all.

**Comment by kconner** [comment]
I think let's leave it out and keep the library small.
